network/network_linux_test.go

```diff
@@ -0,0 +1,98 @@
+package network
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/Azure/azure-container-networking/platform"
+)
+
+func TestAddDNSServersWithErr(t *testing.T) {
+	nm := &networkManager{
+		plClient: platform.NewMockExecClient(true),
+	}
+	_, err := nm.addDNSServers("invalid", []string{""})
+	if err == nil {
+		t.Errorf("Expected error")
+	}
+}
+
+func TestAddDNSServers(t *testing.T) {
+	testCases := []struct {
+		desc      string
+		osversion string
+		ifName    string
+		servers   []string
+		cmd       string
+		err       error
+	}{
+		{
+			desc:      "empty server list",
+			osversion: Ubuntu22,
+			ifName:    "azure0",
+			servers:   []string{},
+			cmd:       "",
+			err:       nil,
+		},
+		{
+			desc:      "one server address",
+			osversion: Ubuntu22,
+			ifName:    "azure0",
+			servers:   []string{"8.8.8.8"},
+			cmd:       "resolvectl dns azure0 8.8.8.8",
+			err:       nil,
+		},
+		{
+			desc:      "multiple server addresses",
+			osversion: Ubuntu22,
+			ifName:    "azure0",
+			servers:   []string{"1.1.1.1", "bing.com"},
+			cmd:       "resolvectl dns azure0 1.1.1.1 bing.com",
+			err:       nil,
+		},
+		{
+			desc:      "empty server list",
+			osversion: "16.0.0",
+			ifName:    "azure0",
+			servers:   []string{},
+			cmd:       "",
+			err:       nil,
+		},
+		{
+			desc:      "one server address",
+			osversion: "16.0.0",
+			ifName:    "azure0",
+			servers:   []string{"8.8.8.8"},
+			cmd:       "systemd-resolve --interface azure0 --set-dns 8.8.8.8",
+			err:       nil,
+		},
+		{
+			desc:      "multiple server addresses",
+			osversion: "16.0.0",
+			ifName:    "azure0",
+			servers:   []string{"1.1.1.1", "bing.com"},
+			cmd:       "systemd-resolve --interface azure0 --set-dns 1.1.1.1 --set-dns bing.com",
+			err:       nil,
+		},
+	}
+	for _, tc := range testCases {
+		t.Log("Running: ", tc.desc, " for ", tc.osversion)
```

Comment:
Please use `t.Run` for this so we can get the benefits of subtests. The name of the subtest can just be `tc.desc`, or you can concatenate the `tc.osversion` as well if you want.